lib/components/common/Tour.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { useState } from 'react';
+import Joyride, { ACTIONS, EVENTS, STATUS, CallBackProps, Step, TooltipRenderProps } from 'react-joyride';
+import { Box, Typography, Button, IconButton } from '@mui/material';
+import Checkbox from './CheckBox';
+import { TutorialCloseIcon } from '../../icons';
+import { vars } from '../../theme/variables';
+import { useUIContext } from '../../contexts/ui/UIContext';
+
+const { baseWhite, gray300, gray500, gray600, primary600, primary700, tutorialOverlayColor, tooltipBoxShadow, gray100, gray900 } = vars;
+
+const Tooltip = ({
+    continuous,
+    index,
+    isLastStep,
+    step,
+    backProps,
+    closeProps,
+    primaryProps,
+    tooltipProps,
+    skipProps,
+    size
+}: TooltipRenderProps) => {
+    const {
+        styles: {
+            tooltip: tooltipStyle,
+            tooltipTitle,
+            tooltipContent,
+            tooltipFooter
+        },
+        hideBackButton,
+        hideCloseButton,
+        hideFooter
+    } = step;
+
+    const isChecked = JSON.parse(localStorage.getItem('isCheckboxChecked') || 'false');
+    const [checked, setChecked] = useState(isChecked);
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setChecked(event.target.checked);
+        localStorage.setItem('isCheckboxChecked', JSON.stringify(event.target.checked));
```

Comment:
suggestions (non-blocking): Every time you see yourself typing the same hardcoded string multiple times in the code, consider replace it by a variable.

suggestion (non-blocking): isCheckboxChecked doesn't seem like a good name to be, I would prefer something like 'skipTutorial'/'skipTour'

Reply:
Sorry, I didn't understand this point as I don't see any code hardcoded multiple times, can you elaborate on this, please @afonsobspinto

Reply:
I was referring to the string "isCheckboxChecked" being hardcoded in the code.

Here you are updating the localStorage with a new value for the key "isCheckboxChecked". And later in the code you will need to read that value. To do so, you ask the localStorage for the value of key "isCheckboxCheked" but if by mistake you type the string incorrectly you will end up with problems not so easy to debug; but if instead you create a variable:

`const localStorageTourKey = "isCheckboxChecked"`

and use it instead you avoid any of those problems

Reply:
Is it clear @Aiga115?

Reply:
yes, it's clear, where can I store this variable, in models.ts? @afonsobspinto

Reply:
I would say https://github.com/MetaCell/cde-mapper/blob/develop/lib/settings.ts